unlock-app/src/components/content/events-collection/AddEventsToCollectionDawer.tsx

```diff
@@ -57,17 +57,83 @@ export default function AddEventsToCollectionDrawer({
 
   const [eventUrl, setEventUrl] = useState<string>('')
   const [eventSlug, setEventSlug] = useState<string>('')
-  const { data: event, isLoading: isLoadingEvent } = useEvent({
+  const {
+    data: event,
+    isLoading: isLoadingEvent,
+    error: eventDetailsError,
+  } = useEvent({
     slug: eventSlug,
   })
   const [addMethod, setAddMethod] = useState<AddMethod>(null)
   const [isUrlValid, setIsUrlValid] = useState(false)
   const [isEventSelected, setIsEventSelected] = useState(false)
   const [isDuplicate, setIsDuplicate] = useState(false)
-  const [hasCheckedUrl, setHasCheckedUrl] = useState<boolean>(false)
+  const [errorMessage, setErrorMessage] = useState<string | null>(null)
 
   const [isSubmitting, setIsSubmitting] = useState<boolean>(false)
 
+  const [debouncedEventUrl, setDebouncedEventUrl] = useState(eventUrl)
+
+  const validateUrl = (url: string) => {
+    const baseUrls = [
+      'https://staging-app.unlock-protocol.com/event/',
+      'https://app.unlock-protocol.com/event/',
```

Comment:
I think this is not the right approach because adding an event from staging will fail in prod and vice versa.
I think you can check the URL based on the environment?

Reply:
true! fixed!